sources/packages/web/src/views/student/NoticeOfAssessment.vue

```diff
@@ -0,0 +1,35 @@
+<template>
+  <Card class="p-m-4">
+    <template #content>
+      <formio formName="noticeofassessment" :data="initialData"></formio>
+    </template>
+  </Card>
+</template>
+
+<script lang="ts">
+import formio from "../../components/generic/formio.vue";
+import { onMounted, ref } from "vue";
+import ApiClient from "../../services/http/ApiClient";
+
+export default {
+  components: { formio },
+  props: {
+    applicationId: {
+      type: Number,
+      required: true,
+    },
+  },
+  setup(props: any) {
+    // Hooks
+    const initialData = ref({});
+    onMounted(async () => {
+      initialData.value = await ApiClient.Application.getNOA(
```

Comment:
I think, you missed the service and you are directly calling the APIclient, actually, we are following the method like 'vue call a service , and the service will call the APIclient

Reply:
But I had hard time finding a service for application, so I had to use StudentService. That's the reason I went via the API Client

Reply:
I agree with @ann-aot, yes, please create a service and use it.

Reply:
ya @guru-aot , you can create a sep service for applications or of it student-specific, you can add in student service